src/views/TimeOutWrapper/index.tsx

```diff
@@ -1,38 +1,152 @@
+import useInterval from 'hooks/useInterval';
 
 type TimeOutWrapperProps = {
   children: React.ReactNode;
 };
 
-const sessionTimeout = { minutes: 14 };
-
-const registerExpire = async (authService: any, lastActiveAt: number) => {
-  const tokenManager = await authService.getTokenManager();
-  // clear the old listener so we don't register millions of them
-  tokenManager.off('expired');
-  tokenManager.on('expired', (key: any) => {
-    const activeSessionWindow = DateTime.local()
-      .minus(sessionTimeout)
-      .toMillis();
-    if (lastActiveAt > activeSessionWindow && key === 'accessToken') {
-      tokenManager.renew(key);
-    }
-  });
-};
-
+/**
+ * As of okta-react 3.0.1, autoRenew on the Security component does
+ * not work. We need to manually renew the session and access token.
+ */
 const TimeOutWrapper = ({ children }: TimeOutWrapperProps) => {
+  const dispatch = useDispatch();
+  const { authState, authService } = useOktaAuth();
+  const { t } = useTranslation();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [timeRemainingArr, setTimeRemainingArr] = useState([0, 'second']);
   const lastActiveAt = useSelector(
     (state: AppState) => state.auth.lastActiveAt
   );
 
-  const { authService }: { authService: any } = useOktaAuth();
-  registerExpire(authService, lastActiveAt);
+  const oneSecond = Duration.fromObject({ seconds: 1 }).as('milliseconds');
+  const fiveMinutes = Duration.fromObject({ minutes: 5 }).as('milliseconds');
+
+  const registerExpire = async () => {
+    const tokenManager = await authService.getTokenManager();
+    // clear the old listener so we don't register millions of them
+    tokenManager.off('expired');
+    tokenManager.on('expired', (key: any) => {
+      const activeSessionWindow = DateTime.local()
+        .minus({ minutes: 14 })
+        .toMillis();
+      if (lastActiveAt > activeSessionWindow) {
+        tokenManager.renew(key);
+      } else {
+        authService.logout('/login');
+      }
+    });
+  };
+
+  const formatSessionTimeRemaining = (
+    timeRemaining: number
+  ): [number, string] => {
+    const seconds = Math.floor(
+      Duration.fromMillis(timeRemaining).as('seconds')
+    );
+
+    const minutes = seconds / 60;
+    // Using Math.ceil() for greater than one minute
+    // 299 seconds = 4.983 minutes, but should still display 5 minutes
+    // Using Math.floor() for less than one minute
+    // 59 seconds = .983 minutes, Using floor so minutes is 0 to display 59 secounds
+    const wholeMinutes = minutes > 1 ? Math.ceil(minutes) : Math.floor(minutes);
+
+    return wholeMinutes > 0
+      ? [wholeMinutes, 'minute']
+      : [Math.floor(seconds), 'second'];
+  };
+
+  const handleModalExit = async () => {
+    setIsModalOpen(false);
+    const tokenManager = await authService.getTokenManager();
+    await tokenManager.renew('accessToken');
+    await tokenManager.renew('idToken');
+    dispatch(updateLastActiveAt);
+  };
+
+  useInterval(
```

Comment:
These different useIntervals could maybe use a comment for which use case they are solving. For example, I think this one handles the interval for updating the text once the modal is open, so the comment could just be: `Count down remaining time and update text when modal is open`.